code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -353,6 +351,28 @@
 	..()
 	base_icon_state = icon_state
 
+/obj/item/weapon/reagent_containers/food/drinks/attack_ghost(mob/dead/observer/user)
+	if(!src || !user)
+		return
+	if(get_dist(src, user) > 1)
+		return
+	if(reagents?.has_reagent(ECTOPLASM))
+		if(!is_open_container())
+			to_chat(user, "<span class='warning'>You can't, [src] is closed.</span>")
+			return
+
+		else if(!reagents?.total_volume)
+			to_chat(user, "<span class='warning'>[src] is empty.<span>")
+			return
+
+		else
+			//user.delayNextAttack(1 SECONDS) Since humans can rapid-drink, we'll leave this commented out for now.
+			to_chat(user, "<span  class='notice'>You swallow a gulp of [src].</span>")
+			reagents.remove_any(gulp_size)
+			playsound(user.loc,'sound/items/drink_ghost.ogg', rand(10,50), 1)
+	else
+		to_chat(user, "<span class='notice'>You pass right through [src].</span>")
```

Comment:
`through \the [src]` - use this everywhere

Reply:
I thought `\the` doesn't do anything.

Reply:
Whether or not [src] is given "the" depends on whether or not the title is improper. So for example, you kick "the" mouse, the gets added automatically. But what about if the object could be proper or improper? You kick Dessert - no article. \the is flexible because it will determine whether it is appropriate to add the article or not.

Mostly useful since admins can rename stuff frequently.

Reply:
That's fine then